thunder/core/rematerialization.py

```diff
@@ -332,6 +332,8 @@ def add_edge(src, dst, capacity):
     def get_weight(var):
         if isinstance(var, TensorProxy):
             return WEIGHT * var.dtype.bytes
+        elif isinstance(var, NumberProxy):
+            return WEIGHT * 0.1
```

Comment:
Why don't we just set the weight to either zero to always cut or to inf to never cut? I kind of see no point in "rematerializing" NumberProxies at all...

Reply:
The problem I met is as [the trace above](https://github.com/Lightning-AI/lightning-thunder/pull/425#issue-2300477976), the seed/offset(which are NumberProxy and not constant values, their values depends on runtime) need to be passed to the backward to generate the random number instead of passing the tensor `t1` to the backward func, so for this case I need a small weight for NumberProxy. Probably zero is a better weight for NumberProxy?

Reply:
In this context - yes! We are not harming this way, are we?

Reply:
I think not, let me check in case

Reply:
I think we should not, because the cutgraph replicates the comp graph with each input/output being independent. In other words, if we have a graph and we have a cut, and if we modify the graph with new inputs/outputs, the resulting cut should be a superset of the previous cut.